File: MyLanguageWeb/src/main/resources/static/js/classes/dto/entity/workout/types/statistic/workout_answers_statistic_response_dto.js
export class WorkoutAnswersStatisticResponseDTO {
    #numberOfAnswers;
    #numberOfTrueAnswers;
    #numberOfFalseAnswers;
    #successRate;

    constructor(json) {
        if (json) {
            this.#numberOfAnswers = json["number_of_answers"];
            this.#numberOfTrueAnswers = json["number_of_true_answers"];
            this.#numberOfFalseAnswers = json["number_of_false_answers"];
            this.#successRate = json["success_rate"];
        }
    }

    getNumberOfAnswers() {
        return this.#numberOfAnswers;
    }

    getNumberOfTrueAnswers() {
        return this.#numberOfTrueAnswers;
    }

    getNumberOfFalseAnswers() {
        return this.#numberOfFalseAnswers;
    }

    getSuccessRate() {
        return this.#successRate;
    }
}

File: MyLanguageWeb/src/main/java/ru/herooo/mylanguageweb/dto/entity/workout/types/statistic/WorkoutAnswersStatisticResponseDTO.java
package ru.herooo.mylanguageweb.dto.entity.workout.types.statistic;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import com.fasterxml.jackson.databind.annotation.JsonSerialize;
import ru.herooo.mylanguageweb.json.l.LongDeserializer;
import ru.herooo.mylanguageweb.json.l.LongSerializer;

public class WorkoutAnswersStatisticResponseDTO {
    @JsonSerialize(using = LongSerializer.class)
    @JsonDeserialize(using = LongDeserializer.class)
    @JsonProperty("number_of_answers")
    private long numberOfAnswers;

    @JsonSerialize(using = LongSerializer.class)
    @JsonDeserialize(using = LongDeserializer.class)
    @JsonProperty("number_of_true_answers")
    private long numberOfTrueAnswers;

    @JsonSerialize(using = LongSerializer.class)
    @JsonDeserialize(using = LongDeserializer.class)
    @JsonProperty("number_of_false_answers")
    private long numberOfFalseAnswers;

    @JsonProperty("success_rate")
    private double successRate;


    public long getNumberOfAnswers() {
        return numberOfAnswers;
    }

    public void setNumberOfAnswers(long numberOfAnswers) {
        this.numberOfAnswers = numberOfAnswers;
    }

    public long getNumberOfTrueAnswers() {
        return numberOfTrueAnswers;
    }

    public void setNumberOfTrueAnswers(long numberOfTrueAnswers) {
        this.numberOfTrueAnswers = numberOfTrueAnswers;
    }

    public long getNumberOfFalseAnswers() {
        return numberOfFalseAnswers;
    }

    public void setNumberOfFalseAnswers(long numberOfFalseAnswers) {
        this.numberOfFalseAnswers = numberOfFalseAnswers;
    }

    public double getSuccessRate() {
        return successRate;
    }

    public void setSuccessRate(double successRate) {
        this.successRate = successRate;
    }
}
